src/libsyntax/parse/parser.rs

```diff
@@ -6172,6 +6174,45 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_expansion(&mut self, kind: ExpansionKind) -> PResult<'a, Expansion> {
```

Comment:
Makes sense -- I'll move it to `expand.rs`. Do you think it should still be a method of `Parser`?